compaction/compaction.cc

```diff
@@ -1105,7 +1110,9 @@ class regular_compaction : public compaction {
                 std::move(trace),
                 sm_fwd,
                 mr_fwd,
-                unwrap_monitor_generator());
+                unwrap_monitor_generator(),
+                default_sstable_predicate(),
+                &_histogram);
```

Comment:
@raphaelsc why is this const?

Reply:
probably it made sense initially, but now I see that even scrub_compaction mutates its internal member in the reader returned.